llvm/lib/Target/SystemZ/SystemZOperators.td

```diff
@@ -1067,6 +1067,15 @@ def vsplat_imm_eq_1 : PatFrag<(ops), (build_vector), [{
 }]>;
 def z_vzext1 : PatFrag<(ops node:$x), (and node:$x, vsplat_imm_eq_1)>;
 
+def vsplat_imm_eq_s32_min : PatFrag<(ops), (build_vector), [{
+  APInt Imm;
+  return ISD::isConstantSplatVector(N, Imm) && Imm.getSExtValue() == -32768;
```

Comment:
Ah, right.  You can probably generalize this function to determine the type (from `N`) and check for the type's minimum/maximum values programmatically.